### ISSUE TITLE
Make the final challenge in the game a planet wide "colossal boss" that can be seen from space.

### ISSUE THREAD

**Issue by sup3r87** [Anuken/Mindustry-Suggestions#3735]
### Describe the content or mechanics you are proposing.

I'm going to start this suggestion off with the obvious; this is a very.. _weird_, and very outlandish/ridiculous idea. The reason I'm typing it here at all is because it occurred in my imagination and I don't want it to go to waste. Alright, enough fluff.

The TL;DR of this suggestion is to make the **final challenge** of the game, whatever planet it may be on, be facing off against a massive "unit", with a universal health bar (shared in the entire game rather than map specific), that slowly "walks" across the planet, and you have to build turrets and infrastructure in sectors you've captured around the point where the unit's leg is going to slam down, since the leg is weaponized and might spawn units.

Here's the more detailed suggestion:

BASIC MECHANICS
The conditions for this unit to spawn would be to fully capture all the campaign centered sectors (campaign sector type means PLT, ruinous shores etc), with around 20 or so other ones. These conditions would be listed like how conditions are listed for sectors. Only instead of the sector on the tech tree, you would see a unit symbol, and an attached warning about the boss.
When the "boss" spawns, it spawns on a random spot on the planet. The boss would then "wander" towards sectors that the player has captured, possibly being more drawn to denser clusters of captured sectors.

STEPPING/HEALTH
The big thing that makes this "boss" unlike a normal unit is that it can be viewed from the planet map. The boss would have 6? 4? around that number of legs, a bit like an arachnid unit. Each leg would be slammed down on one of the sector hexagon tiles.
Each sector that the leg is slammed down on has the leg visible when the player enters that sector; if the player enters any sector with a leg on it, somewhere in the sector would be a massive metallic leg that stretches into the sky. The leg would get more transparent higher up to prevent it obscuring blocks.
Damaging this leg will damage the boss's health, which is displayed ingame and in the planet map. Any legs can be damaged to deal damage to the entire boss.

DEFENSE
This boss will slowly crawl across the planet, and I mean *really* slowly. One leg on the boss will move every, let's say, 10 minutes. And each spot one of the legs is going to step on next is marked on the map. Inside sectors that have yet to have a leg slammed down onto, there'll be a large drop-zone ish area that indicates where the leg is going to slam down. **This is how the player fights the boss**; by rapidly building factories and turrets on new or captured tiles, so that when the leg slams down, they can deal damage to the boss.
When a leg is lifted off a sector, the leg will obviously shoot into the sky as the leg has been, well, lifted off the sector.

OFFENSE
This boss isn't just a DPS tester though, far from it; each foot will have powerful turrets mounted on it, so you have to add walls and defenses like when regular units attack. These turrets on the feet would be considerably powerful to strain the defenses, but not so powerful they rip through defenses after a while. Only defenses that can't technologically complete and/or are badly made can fail.
Each foot will also have a few factories on it to produce units to make the combat more interesting, and less of a single target DPS fest. Maybe each foot can have a different degree of turrets vs unit production. Maybe even different types of damage (lasers, fire, bullets, emps).

DEATH
The boss has to die. I'm not really sure what would be a better death sequence for the boss; for it to give up and fly out into space from the planet, or for it to come crashing down and for the "body" to slam into a sector and completely change the terrain.
A funny idea is, when the body slams down onto a sector, the rubble ruin left in the sector has mineable materials that can't be mined anywhere else, and have little, yet unique uses. Kind of like phase fabric.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

This "boss" would add a huge emphasis on more "grand strategy" and would provide a very cinematic/spectacular end to the game. Having to rapidly build in sector by sector, plan as fast as the boss slowly moves it's legs, and possibly redesign sectors you are in would be VERY different and unique compared to regular mindustry.. and is why I stated at the beginning that this suggestion is very outlandish.

Take the suggestion as an info dump. I know that a LOT would have to go into designing something like this, and it would be a task on the size of making a half of tantros. It was a fun idea that I wanted to put out there instead of leave in my head.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by RFAC-Productions** [comment]
We ironed out the flaws in terms of coding, and we're making it a tripod for it to go easier on the coding in terms of DPS, and determined its size and how it'll spawn. It'll spawn under these conditions:
+ 50+ Captured Sectors
> All Main Sectors Captured
> Any additional sector
+ Research Tree Completed
There will be a sector in the north that will have a giant door, which is heavily guarded.
Now, you might be tginking, "But it doesn't fit with Mindustry." 
Besides being a tower defense + factory game. It is also a war game, as it is a war between factions, Sharded vs Crux, you vs the enemy. Tldr, a strategy game where you deploy tactics, offenses and defenses, to get around and conquer the enemy. 
Since you've taken so many sectors from them, they have to do a last ditch effort to get rid of you, Sharded, by releasing the boss. It is a military tactic, like Fat Man and Little Boy.
We named this Tripod Bos "Stride," because it is marching on, producing units to conquer you, a giant factory.

**Comment by RFAC-Productions** [comment]
We also came up with Erekir and Tantros endgame bosses, "Chilopod" and "Trench" respectively. We don't know what the units if Tantros would look like, but for Erekir, we settled on a giant Centipede/Worm that has its segments show up on a sector. It has a familiar concept as Stride. But we might need to bake the idea a bit more and refine it to give you full details on Chilopod.

**Comment by Noterry** [comment]
Attacking just the legs is boring, so missile pads should help with destroying the boss. Might as well be an exclusive structure _just_ for this fight. It also cannot be used anymore once the boss fight ends.
TLDR Missile pads are pretty much launch pads but offensive, it's basically missile artillery and unlike ripples and foreshadows, it has planetary range. You have a few options of which part of the boss is your target. _To keep it from being a "Spam Missiles agaisnt boss simulator" it will be limited to 1-2 missile pads._ The size will range from 4x4-6x6 tiles. It will take around 30 seconds or more to launch a missile and a few more to hit the boss.

Let's say there's 3 available targets: Weapons, Legs and the Main Body.
Targeting the weapons will halt it's weapons from shooting your frontlines, however it can only shoot a _specific_ weapon. It will help with prioritizing weapons so the legs stepping on your sectors will be the only problem.
Legs, targeting the legs will just help with your frontlines. Once a leg is destroyed, the said leg cannot be targeted anymore.
Main body, targeting it should do "aoe" damage to all legs. Somewhere 0.25-0.33x damage to all. After a few hits or screw it and make it rng based will halt the boss from moving. Though it won't stun the weapons.

Will need atleast 250-500 items for each missile to be launched, there would be missile types depending on resources used.

**Comment by mega2335** [comment]
Nice idea

**Comment by Manymano** [comment]
What if Strider becomes _invulnerable (force field) and stop attacking _for a few minutes when you reach a quarter of half of his HP's? That would let you some time to recover.

**Comment by OnyxFinger** [comment]
wtf is going on there
